lib/wise_homex/json_parser.ex

```diff
@@ -256,13 +259,22 @@ defmodule WiseHomex.JSONParser do
   end
 
   defp set_struct_attribute(struct, key, value) do
-    if Map.has_key?(struct, key) do
-      %{struct | key => value}
-    else
-      struct
+    %{__struct__: module} = struct
+    fields = module.__changeset__()
```

Comment:
I find these lines quite confusing at first sight. Am I right that they create a new, empty struct of the same type, and then sets `fields` to the empty changeset for this empty struct?

Then afterwards this empty changeset is used to find the fields that are of type `[:date, :utc_datetime]` and then converting these to dates.

A suggestion: Put these lines into a separate function that has a descriptive name, and a good comment too.

Reply:
No, `fields` contains a map of `{key, ecto_type}`.
Agree to put it in a separate function.

Reply:
Fixed in 77025602baaf9a0366655aba754770fe9da58ace